src/Sulu/Bundle/MediaBundle/Controller/MediaStreamController.php

```diff
@@ -58,6 +63,20 @@ public function getImageAction(Request $request)
 
             list($id, $format) = $this->getCacheManager()->getMediaProperties($url);
 
+            $lockingKey = sprintf('media-cache_%s_%s', $id, $format);
+
+            $lock = $this->createLock($lockingKey);
+
+            if ($lock && !$lock->acquire()) {
+                // wait for releasing of lock
+                if ($lock->acquire(true)) {
```

Comment:
Just reading the documentation, so I am not sure about this, but wouldn't this suffice:

```php
if ($lock && !$lock->acquire(true)) {
    $lock->release();
    return new RedirectResponse($request->getUri());
}
```

Resp. why isn't that enough?

Reply:
```php
$lock->acquire(true)
```

will waid until the other process release it and then return true so instead of doing a redirect it will regenerated the image which was generated before.

Thats why first we do a none blocking request for the lock if thats false we know somebody else is generating the image next we do a block request just to wait until that process did successfully generated the image and so release and do a redirect then.

Reply:
Okay, now I understand it :+1: